import sys
import os
myDir = os.getcwd()
sys.path.append(myDir)


from Controllers.CreateProductController import CreateProductController
from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_CreateProduct(object):

    def __init__(self):
        self.create_product_controller = CreateProductController(self)

    def setupUi(self, CreateProduct):
        CreateProduct.setObjectName("CreateProduct")
        CreateProduct.resize(400, 300)
        self.layoutWidget = QtWidgets.QWidget(CreateProduct)
        self.layoutWidget.setGeometry(QtCore.QRect(50, 20, 302, 192))
        self.layoutWidget.setObjectName("layoutWidget")
        self.gridLayout = QtWidgets.QGridLayout(self.layoutWidget)
        self.gridLayout.setContentsMargins(0, 0, 0, 0)
        self.gridLayout.setObjectName("gridLayout")
        self.label = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(20)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.gridLayout.addWidget(self.label, 0, 0, 1, 2)
        self.label_2 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.gridLayout.addWidget(self.label_2, 1, 0, 1, 1)
        self.input_cod = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_cod.setFont(font)
        self.input_cod.setObjectName("input_cod")
        self.gridLayout.addWidget(self.input_cod, 1, 1, 1, 1)
        self.label_3 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.gridLayout.addWidget(self.label_3, 2, 0, 1, 1)
        self.input_name = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_name.setFont(font)
        self.input_name.setObjectName("input_name")
        self.gridLayout.addWidget(self.input_name, 2, 1, 1, 1)
        self.label_4 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.gridLayout.addWidget(self.label_4, 3, 0, 1, 1)
        self.input_price = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_price.setFont(font)
        self.input_price.setObjectName("input_price")
        self.gridLayout.addWidget(self.input_price, 3, 1, 1, 1)
        self.label_5 = QtWidgets.QLabel(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.gridLayout.addWidget(self.label_5, 4, 0, 1, 1)
        self.input_category = QtWidgets.QLineEdit(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_category.setFont(font)
        self.input_category.setObjectName("input_category")
        self.gridLayout.addWidget(self.input_category, 4, 1, 1, 1)
        self.btn_create = QtWidgets.QPushButton(self.layoutWidget)
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.btn_create.setFont(font)
        self.btn_create.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.btn_create.setObjectName("btn_create")
        self.gridLayout.addWidget(self.btn_create, 5, 0, 1, 2)

        self.retranslateUi(CreateProduct)
        QtCore.QMetaObject.connectSlotsByName(CreateProduct)
        #--------------------Events--------------------------------------
        self.x = self.btn_create.clicked.connect(lambda:self.create_product_controller.createProduct(self.input_cod.text(),self.input_name.text(),self.input_price.text(),self.input_category.text(),CreateProduct))
        #--------------------End Events---------------------------------

    def retranslateUi(self, CreateProduct):
        _translate = QtCore.QCoreApplication.translate
        CreateProduct.setWindowTitle(_translate("CreateProduct", "Form"))
        self.label.setText(_translate("CreateProduct", "Crear Nuevo Producto"))
        self.label_2.setText(_translate("CreateProduct", "Codigo:"))
        self.input_cod.setPlaceholderText(_translate("CreateProduct", "codigo"))
        self.label_3.setText(_translate("CreateProduct", "Nombre:"))
        self.input_name.setPlaceholderText(_translate("CreateProduct", "nombre"))
        self.label_4.setText(_translate("CreateProduct", "Precio:"))
        self.input_price.setPlaceholderText(_translate("CreateProduct", "precio"))
        self.label_5.setText(_translate("CreateProduct", "Categoria:"))
        self.input_category.setPlaceholderText(_translate("CreateProduct", "categoria"))
        self.btn_create.setText(_translate("CreateProduct", "Crear"))
